// @ant-design/charts/lib/graphs/organizationGraph/customItem.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.registerOrganizationCardNode = void 0;
var g6_1 = __importDefault(require("@antv/g6"));
var constants_1 = require("../constants");
var utils_1 = require("../utils");
// 组织架构图
var registerOrganizationCardNode = function () {
    var defaultIconStyle = {
        width: 12,
        height: 12,
    };
    g6_1.default.registerNode('organization-card', {
        draw: function (cfg, group) {
            if (cfg === void 0) { cfg = {}; }
            var _a = cfg.value, value = _a === void 0 ? {} : _a, nodeCfg = cfg.nodeCfg, markerCfg = cfg.markerCfg;
            var _b = nodeCfg, style = _b.style, _c = _b.padding, padding = _c === void 0 ? 0 : _c, _d = _b.label, label = _d === void 0 ? {} : _d, customContent = _b.customContent;
            var labelStyle = label.style;
            var paddingArray = (0, utils_1.getCssPadding)(padding);
            var size = ((cfg === null || cfg === void 0 ? void 0 : cfg.size) || [100, 30]);
            if (typeof size === 'number')
                size = [size, size];
            var height = 0; // 统计容器总高度，动态设置，宽度不做调整
            var contentWidth = size[0] - paddingArray[1] - paddingArray[3];
            // card box
            var cardStyle = (0, utils_1.getStyle)(style, cfg, group);
            var shape = group.addShape('rect', {
                attrs: __assign(__assign({ x: 0, y: 0, width: size[0], height: size[1] }, constants_1.defaultCardStyle), cardStyle),
                name: 'main-box',
                draggable: true,
            });
            if (value) {
                height += paddingArray[0];
                var createRowItems_1 = function (item, contentWidth, startX, index) {
                    if (index === void 0) { index = 0; }
                    var iconWidth = 0;
                    var rowHeight = [];
                    var keys = ['icon', 'text', 'value'];
                    var getXY = function (type, layoutCfg) {
                        var _a = layoutCfg.fontSize, fontSize = _a === void 0 ? 12 : _a;
                        var x = 0;
                        var y = 0;
                        switch (type) {
                            case 'icon':
                                x = startX;
                                y = height;
                                break;
                            case 'text':
                                x = startX + (contentWidth + iconWidth) / 2;
                                y = item.value ? paddingArray[0] : (size[1] - fontSize) / 2;
                                break;
                            case 'value':
                                x = startX + (contentWidth + iconWidth) / 2;
                                y = item.text
                                    ? paddingArray[0] + rowHeight[1] + constants_1.defaultMargin
                                    : (size[1] - fontSize) / 2;
                                break;
                            default:
                                break;
                        }
                        return { x: x, y: y };
                    };
                    keys.forEach(function (key, keyIndex) {
                        var isIcon = key.startsWith('icon');
                        var shapeStyle = (0, utils_1.getStyle)(labelStyle, cfg, group, key);
                        if (key === 'icon' && item[key]) {
                            iconWidth = shapeStyle.width || 32;
                        }
                        var keyShape = group.addShape(isIcon ? 'image' : 'text', {
                            attrs: __assign(__assign(__assign(__assign({ textBaseline: 'top', textAlign: 'center' }, getXY(key, shapeStyle)), { text: item[key], img: item[key] }), (isIcon ? defaultIconStyle : constants_1.defaultLabelStyle)), shapeStyle),
                            name: key + "-" + index + "-" + keyIndex,
                        });
                        rowHeight.push(keyShape.getBBox().height);
                    });
                    return rowHeight;
                };
                var createItems = function (item, index) {
                    var _a;
                    if (index === void 0) { index = 0; }
                    var itemsHeight = [];
                    if (customContent) {
                        itemsHeight.push((_a = customContent(item, group, {
                            startX: paddingArray[3],
                            startY: height,
                            width: contentWidth,
                        })) !== null && _a !== void 0 ? _a : 0);
                    }
                    else {
                        itemsHeight.push.apply(itemsHeight, createRowItems_1(item, contentWidth, paddingArray[3], index));
                    }
                    height += Math.max.apply(Math, itemsHeight);
                };
                createItems(value);
            }
            shape === null || shape === void 0 ? void 0 : shape.attr('height', Math.max(height + paddingArray[2], size[1]));
            // collapse marker
            if (markerCfg) {
                var _e = shape.getBBox(), shapeWidth = _e.width, shapeHeight = _e.height;
                var _f = typeof markerCfg === 'function' ? markerCfg(cfg, group) : markerCfg, show = _f.show, _g = _f.position, position = _g === void 0 ? 'right' : _g, collapsed = _f.collapsed, markerStyle = _f.style;
                (0, utils_1.createMarker)({
                    show: show,
                    position: position,
                    collapsed: collapsed,
                    style: markerStyle,
                }, group, [shapeWidth, shapeHeight]);
                shape.attr('defaultCollapsed', collapsed);
            }
            return shape;
        },
    }, 'single-node');
};
exports.registerOrganizationCardNode = registerOrganizationCardNode;
