// @ant-design/charts/lib/index.charts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.G2 = exports.Mix = exports.MultiView = exports.Sankey = exports.BidirectionalBar = exports.Gauge = exports.RadialBar = exports.Stock = exports.DualAxes = exports.Sunburst = exports.WordCloud = exports.Waterfall = exports.TinyLine = exports.Violin = exports.TinyColumn = exports.TinyArea = exports.Scatter = exports.Rose = exports.RingProgress = exports.Facet = exports.Radar = exports.Progress = exports.Pie = exports.Heatmap = exports.Liquid = exports.Line = exports.Histogram = exports.Funnel = exports.Column = exports.Bullet = exports.Box = exports.Bar = exports.Area = void 0;
// 从 index import 会有包体积问题
var area_1 = __importDefault(require("./plots/area"));
exports.Area = area_1.default;
var bar_1 = __importDefault(require("./plots/bar"));
exports.Bar = bar_1.default;
var bullet_1 = __importDefault(require("./plots/bullet"));
exports.Bullet = bullet_1.default;
var column_1 = __importDefault(require("./plots/column"));
exports.Column = column_1.default;
var funnel_1 = __importDefault(require("./plots/funnel"));
exports.Funnel = funnel_1.default;
var histogram_1 = __importDefault(require("./plots/histogram"));
exports.Histogram = histogram_1.default;
var line_1 = __importDefault(require("./plots/line"));
exports.Line = line_1.default;
var box_1 = __importDefault(require("./plots/box"));
exports.Box = box_1.default;
var liquid_1 = __importDefault(require("./plots/liquid"));
exports.Liquid = liquid_1.default;
var heatmap_1 = __importDefault(require("./plots/heatmap"));
exports.Heatmap = heatmap_1.default;
var pie_1 = __importDefault(require("./plots/pie"));
exports.Pie = pie_1.default;
var gauge_1 = __importDefault(require("./plots/gauge"));
exports.Gauge = gauge_1.default;
var progress_1 = __importDefault(require("./plots/progress"));
exports.Progress = progress_1.default;
var radar_1 = __importDefault(require("./plots/radar"));
exports.Radar = radar_1.default;
var facet_1 = __importDefault(require("./plots/facet"));
exports.Facet = facet_1.default;
var ringProgress_1 = __importDefault(require("./plots/ringProgress"));
exports.RingProgress = ringProgress_1.default;
var rose_1 = __importDefault(require("./plots/rose"));
exports.Rose = rose_1.default;
var scatter_1 = __importDefault(require("./plots/scatter"));
exports.Scatter = scatter_1.default;
var tinyArea_1 = __importDefault(require("./plots/tinyArea"));
exports.TinyArea = tinyArea_1.default;
var tinyColumn_1 = __importDefault(require("./plots/tinyColumn"));
exports.TinyColumn = tinyColumn_1.default;
var violin_1 = __importDefault(require("./plots/violin"));
exports.Violin = violin_1.default;
var dualAxes_1 = __importDefault(require("./plots/dualAxes"));
exports.DualAxes = dualAxes_1.default;
var tinyLine_1 = __importDefault(require("./plots/tinyLine"));
exports.TinyLine = tinyLine_1.default;
var waterfall_1 = __importDefault(require("./plots/waterfall"));
exports.Waterfall = waterfall_1.default;
var wordCloud_1 = __importDefault(require("./plots/wordCloud"));
exports.WordCloud = wordCloud_1.default;
var sunburst_1 = __importDefault(require("./plots/sunburst"));
exports.Sunburst = sunburst_1.default;
var stock_1 = __importDefault(require("./plots/stock"));
exports.Stock = stock_1.default;
var radialBar_1 = __importDefault(require("./plots/radialBar"));
exports.RadialBar = radialBar_1.default;
var sankey_1 = __importDefault(require("./plots/sankey"));
exports.Sankey = sankey_1.default;
var mix_1 = __importDefault(require("./plots/mix"));
exports.MultiView = mix_1.default;
var mix_2 = __importDefault(require("./plots/mix"));
exports.Mix = mix_2.default;
var bidirectionalBar_1 = __importDefault(require("./plots/bidirectionalBar"));
exports.BidirectionalBar = bidirectionalBar_1.default;
var g2plot_1 = require("@antv/g2plot");
Object.defineProperty(exports, "G2", { enumerable: true, get: function () { return g2plot_1.G2; } });
